draft-ietf-ohai-ohttp.md

```diff
@@ -684,11 +684,19 @@ indicating the content type, and the encapsulated response as the response
 content.  As with requests, additional fields MAY be used to convey information
 that does not reveal information about the encapsulated response.
 
-An Oblivious Gateway Resource that does not receive a response can itself
-generate a response with an appropriate error status code (such as 504 (Gateway
-Timeout); see {{Section 15.6.5 of HTTP}}), which is then encapsulated in the
+An Oblivious Gateway Resource that fails to process the decapsulated request
+or does not receive a response from the Target Resource can itself generate a
+response with an appropriate error status code (such as 400 (Bad Request) or
+504 (Gateway Timeout); see {{Section 15.5.1 of HTTP}} and
+{{Section 15.6.5 of HTTP}}, respectively), which is then encapsulated in the
 same way as a successful response.
```

Comment:
This is a little confused.

We have two cases to talk about here:

1. The gateway failed to process the request in a way that prevents it from encapsulating the response.  Here it would send a cleartext response to the relay.
2. The gateway decrypted the request and so can encapsulate the response, but something else went wrong.  Here it can encapsulate the response.


In the first case, there is no other option.  In the second, the gateway might still decide to send a cleartext response.  However, the relay probably isn't in a good position to do much with a cleartext response - it might treat that as a failure (then see above) - so we might suggest encapsulating a response where possible.

Reply:
I agree with what you said, but I don't see how the proposed text conflicts with it. What specifically about the proposed text do you think needs to change?

Reply:
OK, I'll be more concrete.

Your text says, roughly:

```
if (a: cannot decapsulate) or (b: receives no response from upstream):
  generate a response with an error status code
  encapsulate response
  send encapsulated response
```

However, the encapsulation is not possible in case (a).

My comment was more broadly about how you might structure text:

1. distinguish between decapsulation failure (a) and other failures (b)
2. identify appropriate handling
   a. for case (a)
   b. for case (b), or when a server has case (a) and feels like it; but note the limitations of this

Reply:
This is probably a parsing issue, as that's not what the text says. Failure to process the _decapsulated_ request is not the same as failing to process the encapsulated request, where the latter is decapsulation failure.